src/logic/calculate.js

```diff
@@ -0,0 +1,24 @@
+import operate from './operate';
+
+const calculate = (data, buttonName) => {
+  const operators = ['+', '-', 'X', '÷', '%'];
+  let { total } = data;
+  let { next } = data;
+  const { operation } = data;
+  const library = {
+    '+/-': () => {
+      total *= -1;
+      next *= -1;
+    },
+    '=': () => total,
+    AC: () => { total = 0; },
+  };
+  const isOperation = buttonName => operators.includes(buttonName);
+
+  if (isOperation(buttonName)) {
+    return operate(total, next, operation);
+  }
+  return library[operation];
+};
```

Comment:
This logic needs a complete rework, from what I can see in your PR message some clarification may be in order, let's think of the logic of this file like this:
- We are taking a data object as a parameter with the properties of next, total and operation, this properties are directly related to the buttonName, and they need to be reassigned according to the value of the buttonName, which will be the value given by the user when he clicks on a calculator button, to provide further insight let's take the example provided in the [Pathwright to this milestone](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896094/): 
![image](https://user-images.githubusercontent.com/46115725/73974216-973a9480-48e9-11ea-9276-490900f8edb6.png)
- So to give you a run down, when the user is pressing buttons that are digits you are building your next property as a string so a pseudo-code approach to that would be something like this `if user clicks a button that's a digit then next += buttonName` so let's say you build up the number 25 and then the user presses the `+/-` button, the pseudo-code approach would be like this `if user presses +/- then we set next *= -1`. Another clear example for the logic behind this file would be when the user presses AC so `if the user presses AC then next = null, total = null, operation = null`, so we will have to follow that train of thought when the user presses an operation button what should happen when the user presses and operation after building a number what should the properties values should be every button click should modify the properties of the data object, think through it and try to come up with a solution, does not need to be exact or match every possible escenario right now, but the logic should be present. As the major implementation will be in Milestone 5, once you connect your logic to your components it will make much more sense.